includes/gateways/functions.php

```diff
@@ -327,4 +329,173 @@ function give_get_ordered_payment_gateways( $gateways ) {
 	 * @param array $gateways All the available gateways
 	 */
 	return apply_filters( 'give_payment_gateways_order', $gateways );
-}
\ No newline at end of file
+}
+
+
+/**
+ * Create payment.
+ *
+ * @param $payment_data
+ *
+ * @return bool|int
+ */
+function give_create_payment( $payment_data ) {
+
+	$form_id  = intval( $payment_data['post_data']['give-form-id'] );
+	$price_id = isset( $payment_data['post_data']['give-price-id'] ) ? $payment_data['post_data']['give-price-id'] : '';
+
+	// Collect payment data.
+	$payment_data = array(
+		'price'           => $payment_data['price'],
+		'give_form_title' => $payment_data['post_data']['give-form-title'],
+		'give_form_id'    => $form_id,
+		'give_price_id'   => $price_id,
+		'date'            => $payment_data['date'],
+		'user_email'      => $payment_data['user_email'],
+		'purchase_key'    => $payment_data['purchase_key'],
+		'currency'        => give_get_currency(),
+		'user_info'       => $payment_data['user_info'],
+		'status'          => 'pending',
+		'gateway'         => 'paypal',
+	);
+
+	// Record the pending payment.
+	return give_insert_payment( $payment_data );
+}
+
+/**
+ * Build paypal url
+ *
+ * Note: For internal use only.
+ *
+ * @param int   $payment_id   Payment ID
+ * @param array $payment_data Array of payment data.
+ *
+ * @return mixed|string
+ */
+function _give_build_paypal_url( $payment_id, $payment_data ) {
```

Comment:
Can we more this into the `gateways/paypal-standard.php` file? Also, the underscore seems not very useful unless we implement it across the project. I'd say lost the leading `_` of the function.